cosmwasm/register-its-token.js

```diff
@@ -0,0 +1,128 @@
+require('dotenv').config();
+const env = process.env.ENV;
+const mnemonic = process.env.MNEMONIC;
+
+const { ethers } = require('hardhat');
+const { Contract, getDefaultProvider, constants: { AddressZero } } = ethers;
+const info = require(`../axelar-chains-config/info/${env}.json`);
+const tokenInfo = require(`../axelar-chains-config/info/tokens-${env}.json`);
+const IInterchainTokenService = require('@axelar-network/interchain-token-service/artifacts/contracts/interfaces/IInterchainTokenService.sol/IInterchainTokenService.json');
+const ITokenManager = require('@axelar-network/interchain-token-service/artifacts/contracts/interfaces/ITokenManager.sol/ITokenManager.json');
+const IInterchainToken = require('@axelar-network/interchain-token-service/artifacts/contracts/interfaces/IInterchainToken.sol/IInterchainToken.json')
+const fs = require('fs');
+const toml = require('toml');
+const { printInfo, printError, loadConfig, saveConfig } = require('../common');
+const { initContractConfig, prepareWallet, prepareClient, instantiateContract } = require('./utils');
+const { GasPrice } = require('@cosmjs/stargate');
+
+class TokenIterator {
+    constructor() {
+        this.tokenIndex = -1;
+        this.tokenIds = Object.keys(tokenInfo);
+        this.incrementTokenIndex();
+        this.chainIndex = -1;
+    }
+
+    tokenId() {
+        return this.tokenIds[this.tokenIndex];
+    }
+
+    token() {
+        return tokenInfo[this.tokenId()];
+    }
+
+    incrementTokenIndex() {
+        if(this.tokenIndex >= this.tokenIds.length - 1) return false;
+        this.tokenIndex++;
+        this.chainNames = Object.keys(this.token()).slice(0, -1);
+        this.chainIndex = 0;
+        return true;
+    }
+
+    get() {
+        return this.token()[this.chainName()];
+    }
+
+    chainName() {
+        return this.chainNames[this.chainIndex];
+    }
+
+    rpc() {
+        const chainName = this.chainName();
+        return env === 'mainnet' || env === 'testnet' ? RPCs.axelar_bridge_evm.find((chain) => chain.name.toLowerCase() === chainName).rpc_addr : info.chains[chainName].rpc;
+    }
+
+    async getNext() {
+        const previous = this.get();
+        if (previous && previous.supply) delete previous.supply;
+        fs.writeFileSync(`./axelar-chains-config/info/tokens-${env}.json`, JSON.stringify(tokenInfo, null, 2));                 
+        while (true) {
+            this.chainIndex++;
+            if (this.chainIndex >= this.chainNames.length) {
+                if(!this.incrementTokenIndex()) return false;
+            }
+            const chainName = this.chainName()
+            if(chainName === this.token().originChain.origin_chain) continue;
+            const current = this.get();
+            if (!current.registered) {
+                if (current.track) try {
+                    const provider = getDefaultProvider(this.rpc());
```

Comment:
not very readable, use more newlines, return early, avoid deep nested code, add some comments

Reply:
done, check it out and lmk if you find it ok.